### PR TITLE
Draft: Add a comment to PRs on synchronization.

### PR DESCRIPTION
**Pull request by vibraphone**
### Summary

When a PR is opened, create a comment with a link to CDash corresponding to its SHA. When a PR is synchronized (i.e., a push is made to the source branch of the PR), append a new CDash link to the comment.

### IBAMR Pull Request Checklist
- [ ] Does the test suite pass?
- [ ] Was clang-format run?
- [ ] Were relevant issues cited? Please remember to add `Fixes #12345` to close
      the issue automatically if we are fixing the problem.
- [ ] Is this a change others will want to know about? If so, then has a
      changelog entry been added?
- [ ] If new data structures have been added to a class then have they been set
      up appropriately for restarts? If so, ensure that the restart version
      number is incremented.
- [ ] Does this change include a bug fix or new functionality? If so, a new test
      or tests should be added. New tests should run quickly (less than a minute
      in release mode). If possible, an older test should gain a new option so
      that we do not need to compile more test executables.
- [ ] Did you (if your account has permission to do so) set relevant labels on
      GitHub for the pull request?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,6 +1,8 @@
 IBAMR
 =====
 
+:robot:
+
```
Comment (drwells) [inline README.md:5]: Are we going to have `build : passing` here soon?

File: .github/workflows/pr-comment.yml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+            + [&#x1F517; ${{ steps.sha.outputs.sha }}](https://open.cdash.org/index.php?project=IBAMR&filtercount=1&showfilters=0&field1=revision&compare1=63&value1=${{ steps.sha.outputs.sha }})
+      - name: Add/update link comment
+        if: steps.fc.outputs.comment-id != '' && ${{ github.event.pull_request.action == 'synchronize' }}
+        uses: peter-evans/create-or-update-comment@v1
+        id: dashboard_link
+        with:
+          comment-id: ${{ steps.fc.outputs.comment-id }}
+          issue-number: ${{ github.event.pull_request.number }}
+          body: |
+            + [&#x1F517; ${{ steps.sha.outputs.sha }}](https://open.cdash.org/index.php?project=IBAMR&filtercount=1&showfilters=0&field1=revision&compare1=63&value1=${{ steps.sha.outputs.sha }})
```
Comment (drwells) [inline .github/workflows/pr-comment.yml:55]: Some comments would help - is that a color code?

File: README.md (hunk 1 above)
Comment (vibraphone) [reply README.md:5]: If you want a badge in the readme, [that is a separate thing](https://github.com/brennonbrimhall/CBadge). Injecting javascript into the comments here would likely be difficult due to security policies.

File: .github/workflows/pr-comment.yml (hunk 2 above)
Comment (vibraphone) [reply .github/workflows/pr-comment.yml:55]: No, that is a "link" symbol: &#x1F517;

Comment (vibraphone) [reply .github/workflows/pr-comment.yml:55]: I've started experimenting in my fork as debugging this requires merging any `pull_request_target` action to use it. So, I'll close this for now.

**Pull request closed without merging**